cli/flag/flag.go

```diff
@@ -0,0 +1,257 @@
+package flag
+
+import (
+	"time"
+
+	charts "github.com/linkerd/linkerd2/pkg/charts/linkerd2"
+	"github.com/spf13/pflag"
+)
+
+type (
+	// Flag is an interface which describes a command line flag that affects the
+	// Helm Values used to render Helm charts.  This interface allows us to
+	// iterate over flags which have been set and apply their effects to the
+	// Values.
+	Flag interface {
+		Apply(values *charts.Values) error
+		IsSet() bool
+		Name() string
+	}
+
+	// UintFlag is a Flag with a uint typed value.
+	UintFlag struct {
+		name    string
+		Value   uint
+		flagSet *pflag.FlagSet
+		apply   func(values *charts.Values, value uint) error
+	}
+
+	// Int64Flag is a Flag with an int64 typed value.
+	Int64Flag struct {
+		name    string
+		Value   int64
+		flagSet *pflag.FlagSet
+		apply   func(values *charts.Values, value int64) error
+	}
+
+	// StringFlag is a Flag with a string typed value.
+	StringFlag struct {
+		name    string
+		Value   string
+		flagSet *pflag.FlagSet
+		apply   func(values *charts.Values, value string) error
+	}
+
+	// StringSliceFlag is a Flag with a []string typed value.
+	StringSliceFlag struct {
+		name    string
+		Value   []string
+		flagSet *pflag.FlagSet
+		apply   func(values *charts.Values, value []string) error
+	}
+
+	// BoolFlag is a Flag with a bool typed value.
+	BoolFlag struct {
+		name    string
+		Value   bool
+		flagSet *pflag.FlagSet
+		apply   func(values *charts.Values, value bool) error
+	}
+
+	// DurationFlag is a Flag with a time.Duration typed value.
+	DurationFlag struct {
+		name    string
+		Value   time.Duration
+		flagSet *pflag.FlagSet
+		apply   func(values *charts.Values, value time.Duration) error
+	}
+)
+
+// NewUintFlag creates a new uint typed Flag that executes the given function
+// when applied.  The flag is attached to the given FlagSet.
+func NewUintFlag(flagSet *pflag.FlagSet, name string, defaultValue uint, description string, apply func(values *charts.Values, value uint) error) *UintFlag {
+	flag := UintFlag{
+		name:    name,
+		flagSet: flagSet,
+		apply:   apply,
+	}
+	flagSet.UintVar(&flag.Value, name, defaultValue, description)
+	return &flag
+}
+
+// NewInt64Flag creates a new int64 typed Flag that executes the given function
+// when applied.  The flag is attached to the given FlagSet.
+func NewInt64Flag(flagSet *pflag.FlagSet, name string, defaultValue int64, description string, apply func(values *charts.Values, value int64) error) *Int64Flag {
```

Comment:
NIT: Maybe you can consider using type aliases for the apply functions to make the signatures a bit more concise